tests/posix/common/testcase.yaml

```diff
@@ -27,15 +27,15 @@ tests:
     extra_configs:
       - CONFIG_NEWLIB_LIBC=n
       - CONFIG_THREAD_LOCAL_STORAGE=y
-      - CONFIG_MAIN_STACK_SIZE=1152
+      - CONFIG_MAIN_STACK_SIZE=4096
   portability.posix.common.tls.newlib:
     platform_exclude: nsim_sem_mpu_stack_guard ehl_crb lpcxpresso55s06
     filter: TOOLCHAIN_HAS_NEWLIB == 1 and CONFIG_ARCH_HAS_THREAD_LOCAL_STORAGE and
       CONFIG_TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE
     extra_configs:
       - CONFIG_NEWLIB_LIBC=y
       - CONFIG_THREAD_LOCAL_STORAGE=y
-      - CONFIG_MAIN_STACK_SIZE=1152
+      - CONFIG_MAIN_STACK_SIZE=4096
```

Comment:
while I appreciate that the main stack needed to be increased, this seems like a really large increase? Do we have any insight on what the size really needed?

Reply:
I agree - it would be good to know exactly how much space is required, and on what platforms.

Reply:
@cfriedt @dleach02 any idea on how to track the memory usage? per my understanding is that it depends on how many threads are processed in parallally, which means if the system runs faster, this issue may gone.